plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -168,19 +174,24 @@ export class GitlabOrgDiscoveryEntityProvider implements EntityProvider {
       logger: logger,
     });
 
-    const users = paginated<GitLabUser>(options => client.listUsers(options), {
-      page: 1,
-      per_page: 100,
-      active: true,
-    });
+    let groups;
+    let users;
 
-    const groups = paginated<GitLabGroup>(
-      options => client.listGroups(options),
-      {
+    if (client.isSelfManaged()) {
+      groups = paginated<GitLabGroup>(options => client.listGroups(options), {
         page: 1,
         per_page: 100,
-      },
-    );
+      });
+
+      users = paginated<GitLabUser>(options => client.listUsers(options), {
+        page: 1,
+        per_page: 100,
+        active: true,
+      });
+    } else {
+      groups = (await client.listSaasGroups(this.config.group)).items;
+      users = (await client.listSaasUsers(this.config.group)).items;
```

Comment:
Usually when you are using gitlab saas you add users to the root level namespace and sync them with your sso provider.

`group` field in config can be a subgroup because you may want to fetch the org structure only from a subgroup. E.g. in my company we use `<root-org-group>/teams`

IMHO this call should be:

```suggestion
      users = (await client.listSaasUsers(this.config.group.split('/')[0])).items;
```

And we also need to ensure that it does not start with a `/`

Reply:
I believe we have a similar setup. Users are added to the root group with SAML SSO with a Minimal Access role. Additionally, we add users as direct members to various team groups via SAML group linking.

We do not see value in consuming the root group and all users as Backstage entities. We want the team groups and associated members ingested as entities for ownership and to leverage the sign-in resolver etc. within Backstage.

I believe it may be possible to achieve what you need by tweaking the provider config perhaps.

Reply:
Having re-looked at the current code, I think this change does in fact make sense, so thank you for the suggestion!

Reply:
Updated